test_runner/src/main/kotlin/ftl/ios/xctest/ReduceXcTestRunV2.kt

```diff
@@ -0,0 +1,49 @@
+package ftl.ios.xctest
+
+import com.dd.plist.NSArray
+import com.dd.plist.NSDictionary
+import com.dd.plist.NSObject
+import ftl.ios.xctest.common.TEST_CONFIGURATIONS
+import ftl.ios.xctest.common.TEST_TARGETS
+import ftl.ios.xctest.common.getBlueprintName
+import ftl.ios.xctest.common.getTestConfigurations
+import ftl.ios.xctest.common.getTestTargets
+import ftl.ios.xctest.common.setOnlyTestIdentifiers
+
+fun NSDictionary.reduceXcTestRunV2(
+    configuration: String,
+    targets: Map<String, List<String>>
+): NSDictionary = apply {
+    set(
+        TEST_CONFIGURATIONS,
+        getTestConfigurations()
+            .getValue(configuration)
+            .reduceTestConfiguration(targets)
+            .inNSArray()
+    )
+}
+
+private fun NSDictionary.reduceTestConfiguration(
+    targets: Map<String, List<String>>
+): NSDictionary = apply {
+    set(
+        TEST_TARGETS,
+        getTestTargets()
+            .mapNotNull {
+                targets[
+                    it.getBlueprintName()
+                ]?.let { methods ->
+                    it.setOnlyTestIdentifiers(methods)
+                }
```

Comment:
```kotlin
targets[ it.getBlueprintName()]
    ?.let { methods ->it.setOnlyTestIdentifiers(methods)}
```

Reply:
Pls, use GitHub's suggestion for convenience.

Reply:
it does not work well with multilines 😄

Reply:
> it does not work well with multilines 😄

Only for lines that are not changed in PR.